src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -61,16 +65,20 @@ public void storeMessage(Message message) {
             .addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
 
-    for (Entity entity : results.asIterable()) {
-      try {
+    for (Entity entity : results.asIterable())
```

Comment:
Nice! Using a ranged based for is a good idea here especially since you don't need the index of each item in results.

Reply:
This came with the original code but thank you!